File: rb101/rb101-rb109-small_problems/easy5/letter_counter1.rb
def word_sizes(string)
  word_sizes = Hash.new(0)
  string.split.each do |word|
    word_sizes[word.size] += 1
  end
  word_sizes
end

puts word_sizes('Four score and seven.') == { 3 => 1, 4 => 1, 5 => 1, 6 => 1 }
puts word_sizes('Hey diddle diddle, the cat and the fiddle!') == { 3 => 5, 6 => 1, 7 => 2 }
puts word_sizes("What's up doc?") == { 6 => 1, 2 => 1, 4 => 1 }
puts word_sizes('') == {}

# ALTERNATIVE SOLUTION:

# def word_sizes(string)
#   word_sizes = {}
#   string.split.each do |word|
#     if word_sizes[word.size]
#       word_sizes[word.size] += 1
#     else
#       word_sizes[word.size] = 1
#     end
#   end
#   word_sizes
# end

File: js210/js210-small_problems/easy3/letterCounter1.js
function wordSizes(string) {
  const wordsArray = string.split(' ');
  const sizesObj = {};

  wordsArray.forEach(word => {
    let wordSize = word.length;

    if (wordSize === 0) return;

    if (sizesObj[wordSize]) {
      sizesObj[wordSize] += 1;
    } else {
      sizesObj[wordSize] = 1;
    }
  });

  return sizesObj;
}

console.log(wordSizes('Four score and seven.'));                       // { "3": 1, "4": 1, "5": 1, "6": 1 }
console.log(wordSizes('Hey diddle diddle, the cat and the fiddle!'));  // { "3": 5, "6": 1, "7": 2 }
console.log(wordSizes("What's up doc?"));                              // { "2": 1, "4": 1, "6": 1 }
console.log(wordSizes(''));                                            // {}


// ALTERNATIVE SOLUTION:

// function wordSizes(words) {
//   const wordsArray = words.split(' ');
//   const count = {};

//   for (let i = 0; i < wordsArray.length; i += 1) {
//     let wordSize = wordsArray[i].length;
//     if (wordSize === 0) {
//       continue;
//     }

//     count[wordSize] = count[wordSize] || 0;
//     count[wordSize] += 1;
//   }

//   return count;
// }
